repository/src/test/java/org/alfresco/repo/content/StorageClassTest.java

```diff
@@ -0,0 +1,313 @@
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.repo.content;
+
+import org.alfresco.model.ContentModel;
+import org.alfresco.repo.content.filestore.FileContentStore;
+import org.alfresco.repo.security.authentication.AuthenticationComponent;
+import org.alfresco.repo.transaction.RetryingTransactionHelper;
+import org.alfresco.service.cmr.repository.ChildAssociationRef;
+import org.alfresco.service.cmr.repository.ContentService;
+import org.alfresco.service.cmr.repository.ContentWriter;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.NodeService;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.namespace.QName;
+import org.alfresco.service.transaction.TransactionService;
+import org.alfresco.test_category.OwnJVMTestsCategory;
+import org.alfresco.util.BaseSpringTest;
+import org.alfresco.util.GUID;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.mockito.Spy;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import javax.transaction.NotSupportedException;
+import javax.transaction.SystemException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+
+@Category(OwnJVMTestsCategory.class)
+@RunWith(SpringRunner.class) public class StorageClassTest extends BaseSpringTest
+{
+        private static final String DEFAULT_SC = "Default1";
+        private static final String DEFAULT_VALUE = "default";
+        private static final String TEST_NAMESPACE = "TestNameSpace";
+        private TransactionService transactionService;
+        private AuthenticationComponent authenticationComponent;
+
+        private NodeService nodeService;
+        private NodeRef rootNode;
+        @Spy ContentStore mockContentStore;
```

Comment:
The annotation should above the actual field.

Reply:
Fixed